### PR TITLE
Add ByteString.emptyByteString to Java API

### PR DESCRIPTION
**Pull request by raboof**
To work around https://github.com/scala/bug/issues/11509

Refs #26926

### CODE REVIEW

**Review by dwijnand — APPROVED** [review]

**Review by johanandren — APPROVED** [review]

File: akka-actor/src/main/scala-2.13-/akka/util/ByteString.scala [hunk 1]
```diff
@@ -138,6 +138,9 @@ object ByteString {
 
   val empty: ByteString = CompactByteString(Array.empty[Byte])
 
+  /** Java API */
+  val emptyByteString: ByteString = empty
```
Comment (patriknw) [inline akka-actor/src/main/scala-2.13-/akka/util/ByteString.scala]: should we mention it in migration guide for 2.6? not that it's a 2.6 change, but some java users will probably use 2.13 at the same time as changing to 2.6 (if we push that version as the recommended for java users)

Comment (raboof) [reply akka-actor/src/main/scala-2.13-/akka/util/ByteString.scala]: Might make sense, will do.

File: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Migration Guide 2.5.x to 2.6.x
 
+It is now recommended to use @apidoc[akka.util.ByteString]`.emptyByteString()` instead of
+@apidoc[akka.util.ByteString]`.empty()` when using Java because @apidoc[akka.util.ByteString]`.empty()`
+is [no longer available as a static method](https://github.com/scala/bug/issues/11509) in the artifacts built for Scala 2.13.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md]: good

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**